worker/caasunitprovisioner/deployment_worker.go

```diff
@@ -146,7 +146,11 @@ func (w *deploymentWorker) loop() error {
 		if err != nil {
 			return errors.Trace(err)
 		}
-		spec, err := w.broker.Provider().ParsePodSpec(specStr)
+		provider, ok := w.broker.Provider().(caas.ContainerEnvironProvider)
```

Comment:
If we need a ContainerEnvironProvider we should know when we create the worker that we have it

Reply:
fixed